test/helpers/quictestlib.c

```diff
@@ -995,3 +1085,64 @@ int qtest_fault_resize_datagram(QTEST_FAULT *fault, size_t newlen)
 
     return 1;
 }
+
+/* There isn't a public function to do BIO_ADDR_copy() so we create one */
```

Comment:
Adding a BIO_ADDR_copy function would make sense rather than a private function here - there are other contexts where being able to do that would be useful.

Reply:
Note there is a BIO_ADDR_dup which may be usable as an alternative in some contexts but not in others.

But, anyway I will create a public BIO_ADDR_copy function - but I'd rather do that in a follow up PR rather than creep the scope of this one.

Reply:
Not a problem - just make sure an issue is raised to do that and note it should clean this up when that is in place.
I can see the BIO actually being more generally usable than just what you have it set for here - i.e. other people may want to use this for other testing purposes - so it could end up as a standard BIO outside of the testing area.

Reply:
See openssl/project#220